doc/source/building/index.rst

```diff
@@ -174,6 +175,12 @@ your system.
         run a ``.bat`` file for the correct bitness and architecture (e.g., for
         64-bit Intel CPUs, use ``vcvars64.bat``).
 
+        If using a Conda environment while a version of Visual Studio 2019+ is
+        installed that includes the MSVC v142 package (VS 2019 C++ x86/x64
+        build tools), activating the conda environment should cause Visual
+        Studio to be found and the appropriate .bat file executed to set
+        these variables.
```

Comment:
I could be wrong here, but I don't think Conda executes the VS-shipped `.bat` file but rather its own activation script that simply sets `CC` & co. I like the spirit of the added note here though, it's basically "if you're using a conda env, don't worry it will work automatically".

Reply:
Conda has its own script called during environment activation (I believe included via the conda vs2019_win-64 package) that finds the installed Visual Studio's vcvars.bat and then calls it using an appropriate -vcvars_ver option. This is extremely nice since this MSVC environment variable setup seems quite complex and confusing if you start trying to actually drive into it. Miniforge mamba also launches in Windows via its own batch file setup, so you can't use the Visual Studio setup command prompt launcher at the same time, at least directly. From my experience, what I want to convey to other people trying to get this working is that if using conda, it should just work. If it doesn't, the issue is likely that the right MSVC build tools version wasn't installed.

Reply:
Ok, Eureka! Looking through the numpy-dev conda activation scripts in numpy-dev/etc/conda/activate.d/, I see flang_win-64_activate.bat sets environment variable AR=llvm-ar.exe. Unsetting this variable fixes the build! The same script also sets LDFLAGS to add a clang specific option which triggers unrecognized options warnings during the build and sets LD=lld-link.exe, which doesn't seem to matter, but shouldn't be there.

This script is getting added via the "compilers" package entry in environment.yml. This package is required to get the vcvars auto-setup to work (and maybe for other reasons), so it shouldn't be removed. As far as I can tell, there isn't a way to specify package options to get a version of compilers without flang. I don't know if there's some other good way to get conda to keep everything required installed but just not run the flang activation scripts?

Assuming no smoother solutions are on offer, I propose that I delete this proposed section and instead add a comment in both the "Building from source to use NumPy" and "Building from source for NumPy development" sections recommending unsetting AR, LDLAGS, and LD before running spin build on Windows.